hack/sast.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+#
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+root_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." &> /dev/null && pwd )"
+
+gosec_report="false"
+gosec_report_parse_flags=""
+
+parse_flags() {
+  while test $# -gt 1; do
+    case "$1" in
+      --gosec-report)
+        shift; gosec_report="$1"
+        ;;
+      *)
+        echo "Unknown argument: $1"
+        exit 1
+        ;;
+    esac
+    shift
+  done
+}
+
+parse_flags "$@"
+
+echo "> Running gosec"
+gosec --version
+if [[ "$gosec_report" != "false" ]]; then
+  echo "Exporting report to $root_dir/gosec-report.sarif"
+  gosec_report_parse_flags="-track-suppressions -fmt=sarif -out=gosec-report.sarif -stdout"
+fi
+
+# Gardener uses code-generators https://github.com/kubernetes/code-generator and https://github.com/protocolbuffers/protobuf
```

Comment:
```suggestion
# The shoot-rsyslog-relp extension uses code-generators https://github.com/kubernetes/code-generator and https://github.com/protocolbuffers/protobuf
```

Reply:
Nit: We don't have protobuf in extensions.

Reply:
But in general instead of copying this script file on `make tidy` we could call it directly from the hack dir of the Gardener the same way we call other scripts.

Reply:
I'm ok to not copy it locally under hacks. I think it was done for other scripts so that they can be conveniently called from the hack directory as there is no make target for them. 
Btw, It might be useful to extend the original script under g/g so that additional parameters can be passed to it - e.g. projects that depend on it might have additional directories that they want to exclude.

Reply:
> Btw, It might be useful to extend the original script under g/g so that additional parameters can be passed to it - e.g. projects that depend on it might have additional directories that they want to exclude.

Yep, we also want to exclude the gardener/ dir in the registry-cache repo. Ref https://github.com/gardener/gardener-extension-registry-cache/pull/272#discussion_r1824144071